Components/FlightControl/FlightTask.cpp

```diff
@@ -43,8 +46,26 @@ void FlightTask::InitTask()
  */
 void FlightTask::Run(void * pvParams)
 {
-	//TODO: This should probably be true for enter state, although this behavior is dictated by flash state recovery
-    rsm_ = new RocketSM(RS_ABORT, false);
+    //Initialize SPI Flash
+    SPIFlash::Inst().Init();
+
+    //Get the latest state from the system storage
+    SystemState sysState;
+    bool stateReadSuccess = SystemStorage::Inst().Read(sysState);
+
+    if (stateReadSuccess == false) {
+        // Succeded to read state, initialize the rocket state machine
+		//TODO: Check if we really want to enter the state or not, or if we need to make it selective based on the state
+        rsm_ = new RocketSM(sysState.rocketState, true);
```

Comment:
Whether to re-enter the state is a tough one to figure out: we'll need to test what happens in various power-off stages (eg. full power cut, power brownout, fault, assert, etc.) and make sure it works for every state.

We can also optionally store if the state has been successfully entered to the flash storage, but even if its been entered it may or may not need to enter it again depending on how the board reset

Reply:
Assuming RAM has reset (which it will have), I think re-entering might be the right thing to do. Definitely need to test this though.